hypothesis-python/src/hypothesis/extra/pytestplugin.py

```diff
@@ -128,12 +132,34 @@ def note_statistics(stats):
             gathered_statistics[item.nodeid] = lines
             item.hypothesis_statistics = lines
 
+        # Peturb PRNG so that we can detect e.g. everything calling seed(0)
+        random.seed(random.randrange(2 ** 32))
+        before = hash(random.getstate())
+
         with collector.with_value(note_statistics):
             with with_reporter(store):
                 yield
         if store.results:
             item.hypothesis_report_information = list(store.results)
 
+        # We want to detect any cases where a test (or Hypothesis) seeds or sets the
+        # random state without restoring it afterwards, which pollutes the whole test
+        # suite with correlations.  This check works if more than one test leaks the
+        # same state, which has been true of all our problems so far.  See #1919.
+        after = hash(random.getstate())
+        if before != after:
+            if after in states_after:
+                warnings.warn(
+                    "%r and %r both used the `random` module, and somehow finished "
+                    "with the same global `random.getstate()`, probably thanks to "
+                    "`random.seed()`.\n    This makes Hypothesis test suites much "
```

Comment:
It might be worth making it clear that this is what happens if you call `random.seed` *before* the test is called, not within the test.